graylog2-server/src/main/resources/org/graylog2/plugin/journal/raw_message.proto

```diff
@@ -25,6 +27,8 @@ message JournalMessage {
     optional RemoteAddress remote = 7;
 
     optional bytes payload = 8;
+
+    optional uint32 sequence_nr = 9;
```

Comment:
This means the sequence number is an integer. Wouldn't it be safer to use a long? What happens if the atomic integer wraps to `Integer.MIN_VALUE` on `.incrementAndGet()`? I guess we don't want negative sequence numbers.

Reply:
Good catch. I didn't' use long on purpose to save some space in the journal.
I've added code to avoid wrapping into negative values.